### PR TITLE
SpacesAfterSemicolonFixer - fix loops handling

### PR DESCRIPTION
**Pull request by keradus**
replaces #1683

### CODE REVIEW

File: Symfony/CS/Fixer/Symfony/SpacesAfterSemicolonFixer.php [hunk 1]
```diff
@@ -27,15 +27,26 @@ public function fix(\SplFileInfo $file, $content)
...
+                    $tokens->insertAt($index + 1, new Token(array(T_WHITESPACE, ' ')));
+                }
+            } elseif (
+                isset($tokens[$index + 2])
+                && !$tokens[$index + 1]->equals(array(T_WHITESPACE, ' '))
+                && $tokens[$index + 1]->isWhitespace(array('whitespaces' => " \t"))
+                && !$tokens[$index + 2]->isComment()
+            ) {
+                if ($tokens[$index + 2]->equals(')')) {
+                    $tokens[$index + 1]->clear();
```
Comment (gharlan) [inline Symfony/CS/Fixer/Symfony/SpacesAfterSemicolonFixer.php]: is this the job of this fixer? I'm not sure.
On 2.x it is also done by `ParenthesisFixer`..

Comment (keradus) [inline Symfony/CS/Fixer/Symfony/SpacesAfterSemicolonFixer.php]: on 1.x the fixer already do this, so we should not stop on 1.x

Comment (gharlan) [inline Symfony/CS/Fixer/Symfony/SpacesAfterSemicolonFixer.php]: > on 1.x the fixer already do this

I don't think so: https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/1.12/Symfony/CS/Fixer/Symfony/SpacesAfterSemicolonFixer.php

Can not find any `clear`.

Comment (keradus) [inline Symfony/CS/Fixer/Symfony/SpacesAfterSemicolonFixer.php]: OK, I was suggested by @SpacePossum tests ;)

Comment (SpacePossum) [inline Symfony/CS/Fixer/Symfony/SpacesAfterSemicolonFixer.php]: I wouldn't rely to much on those :P

**Comment by SpacePossum** [comment]
lgtm :+1: ,
thanks :)

**Comment by keradus** [comment]
Thank you @SpacePossum 
This one was tricky indeed

**Comment by gharlan** [comment]
:+1:

**Comment by keradus** [comment]
Thank you @SpacePossum.

**Pull request merged**